src/plugins/visualizations/public/saved_visualizations/saved_visualizations.ts

```diff
@@ -89,6 +92,25 @@ export function createSavedVisLoader(
         visTypes: visualizationTypes.getAliases(),
       });
     }
+
+    // Similar to the default SavedObjectLoader delete(), but doing some extra checking and deletion
+    // of any associated augment-vis saved objects for a particular visualization saved object.
+    async delete(ids: string | string[]): Promise<void> {
```

Comment:
Do we need to move this function to the `SavedObjectLoaderAugmentVis` class?

Reply:
No, because the idea is we're overriding the `delete()` fn of the _visualization_ saved object loader such that we clean up any leftover `augment-vis` saved objs while we're at it. The workflow would be: (1) use vis saved obj loader to delete vis (2) vis saved obj loader will run custom delete() and delete any associated `augment-vis` saved objs in the background